shell/common/node_bindings.cc

```diff
@@ -518,8 +520,13 @@ node::Environment* NodeBindings::CreateEnvironment(
 
   args.insert(args.begin() + 1, init_script);
 
-  if (!isolate_data_)
-    isolate_data_ = node::CreateIsolateData(isolate, uv_loop_, platform);
+  auto* isolate_data = context->GetAlignedPointerFromEmbedderData(
+      ELECTRON_CONTEXT_EMBEDDER_DATA_INDEX);
+  if (!isolate_data) {
+    isolate_data = node::CreateIsolateData(isolate, uv_loop_, platform);
+    context->SetAlignedPointerInEmbedderData(
+        ELECTRON_CONTEXT_EMBEDDER_DATA_INDEX, static_cast<void*>(isolate_data));
+  }
```

Comment:
If we are creating a new environment shouldn't this always create a new isolate data ?

```suggestion
  auto* isolate_data = node::CreateIsolateData(isolate, uv_loop_, platform);
  context->SetAlignedPointerInEmbedderData(
        ELECTRON_CONTEXT_EMBEDDER_DATA_INDEX, static_cast<void*>(isolate_data));
```

Reply:
i don't think so, because we're not creating a new environment every time here (or changing that logic) we're just ensuring that contexts are correctly associated with isolate data. If there's already a context with the correct isolate data, we don't need to change that

Reply:
The code block is currently inside `NodeBindings::CreateEnvironment` which creates an environment every time it is called for a given context, so the newly created context wouldn't have a `node::IsolateData`.

The current check 

```
auto* isolate_data = context->GetAlignedPointerFromEmbedderData(
      ELECTRON_CONTEXT_EMBEDDER_DATA_INDEX);
  if (!isolate_data) {
}
```

will not hold for reasons mentioned in https://github.com/electron/electron/blob/f959fb0c963701c40f309eb8256530f788d5851e/patches/chromium/fix_harden_blink_scriptstate_maybefrom.patch#L26-L40, the failing 32-bit tests confirms this as well. So creating a new isolate data would be an easier route to avoid this and also matches the concept of creating an environment per context. Since in an ideal world both `node::Environment` and `node::IsolateData` are per isolate data structures.

Reply:
addressed!